src/app/clusters/commissioner-control-server/commissioner-control-server.h

```diff
@@ -0,0 +1,99 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommissionerControl {
+
+struct CommissioningApprovalRequest
+{
+    uint64_t requestId;
+    VendorId vendorId;
+    uint16_t productId;
+    NodeId clientNodeId;
+    FabricIndex fabricIndex;
+    Optional<chip::CharSpan> label;
```

Comment:
WHere is the ownership of this data? Suggest making the struct own a char array large enough to be free-standing and owning for the label span to point to.

Reply:
The ownership of this data is the application, who implement the delegate interface 

    virtual CHIP_ERROR HandleCommissioningApprovalRequest(const CommissioningApprovalRequest & request) = 0;
    
 Because this data is only used in the application, not within the Commissioner Control Cluster server